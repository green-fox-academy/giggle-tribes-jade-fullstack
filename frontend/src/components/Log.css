div.log {
  grid-area: 2 / 3 / span 2 / span 1;
}
