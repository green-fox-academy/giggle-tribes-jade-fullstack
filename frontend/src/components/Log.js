import React, { useEffect, useState } from 'react';
import LogEntry from './LogEntry';
import { logEntryService } from '../services/logEntryService';

const Log = () => {
  const [data, setData] = useState([]);

  useEffect(() => {
    setData(logEntryService.readLogEntries());
  }, []);

  return (
    <div className="log">
      <LogEntry header={true} />
      {data.map((entry, i) => (
        <LogEntry key={i} header={false} data={entry} />
      ))}
    </div>
  );
};

export default Log;
