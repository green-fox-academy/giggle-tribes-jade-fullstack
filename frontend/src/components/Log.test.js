import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import Log from './Log';
import moment from 'moment-timezone';

let container;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
  moment.tz.setDefault('UTC');
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('matches snapshot', async () => {
  await act(async () => {
    render(<Log />, container);
  });

  expect(container.innerHTML).toMatchSnapshot();
});
