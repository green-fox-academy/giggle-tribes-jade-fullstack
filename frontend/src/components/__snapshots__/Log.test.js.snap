// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `"<div class=\\"logentry\\"><div class=\\"img\\"></div><section class=\\"head\\"><h2>Construction log:</h2></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>Building Farm Level 1</h3><h3>06:50</h3></header><progress id=\\"subjectProgress\\" value=\\"99\\" max=\\"100\\"></progress></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>Upgrading Academy to Level 2</h3><h3>03:25</h3></header><progress id=\\"subjectProgress\\" value=\\"98\\" max=\\"100\\"></progress></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>Building Troop Level 1</h3><h3>03:25</h3></header><progress id=\\"subjectProgress\\" value=\\"4\\" max=\\"100\\"></progress></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>My Kingdom</h3><h3>03:25</h3></header><p>Troop upgraded!</p></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>My Kingdom</h3><h3>03:25</h3></header><p>You have a new Farm!</p></section></div><div class=\\"logentry\\"><div class=\\"img\\"></div><section><header><h3>Upgrading Troop to Level 2</h3><h3>03:25</h3></header><progress id=\\"subjectProgress\\" value=\\"38\\" max=\\"100\\"></progress></section></div>"`;
